src/main/java/hudson/plugins/timestamper/pipeline/GlobalAnnotator.java

```diff
@@ -77,11 +77,14 @@
         }
         long buildStartTime = build.getStartTimeInMillis();
         String html = text.toString(true);
-        int start;
-        if (html.startsWith("<span class=\"pipeline-new-node\" ")) { // cf. LogStorage.startStep
-            start = html.indexOf('>') + 1;
-        } else {
-            start = 0;
+        int start = 0;
+        // cf. LogStorage.startStep
+        if (html.startsWith("<span class=\"pipeline-new-node\" ", start)) {
+            start = html.indexOf('>', start) + 1;
+        }
+        // cf. AnsiHtmlOutputStream.setForegroundColor
+        if (html.startsWith("<span style=\"color", start)) {
```

Comment:
`else if` syntax more readable, and makes `start` be effectively final.

Reply:
I tried implementing your suggestion with this change:

```java
diff --git a/src/main/java/hudson/plugins/timestamper/pipeline/GlobalAnnotator.java b/src/main/java/hudson/plugins/timestamper/pipeline/GlobalAnnotator.java
index ce8a2d7..4940549 100644
--- a/src/main/java/hudson/plugins/timestamper/pipeline/GlobalAnnotator.java
+++ b/src/main/java/hudson/plugins/timestamper/pipeline/GlobalAnnotator.java
@@ -78,12 +78,11 @@ public final class GlobalAnnotator extends ConsoleAnnotator<Object> {
         long buildStartTime = build.getStartTimeInMillis();
         String html = text.toString(true);
         int start = 0;
-        // cf. LogStorage.startStep
         if (html.startsWith("<span class=\"pipeline-new-node\" ", start)) {
+            // cf. LogStorage.startStep
             start = html.indexOf('>', start) + 1;
-        }
-        // cf. AnsiHtmlOutputStream.setForegroundColor
-        if (html.startsWith("<span style=\"color", start)) {
+        } else if (html.startsWith("<span style=\"color", start)) {
+            // cf. AnsiHtmlOutputStream.setForegroundColor
             start = html.indexOf('>', start) + 1;
         }
         if (html.startsWith("[", start)) {
```

However, it caused the tests to start failing with this exception:

```Finished: SUCCESS expected:<21> but was:<19>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at hudson.plugins.timestamper.pipeline.PipelineTest.globalDecoratorAnnotator(PipelineTest.java:100)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:552)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Looking into it further, I see HTML like this:

```
<span class="timestamp"><b>12:39:46</b> </span><span style="display: none">[2019-07-05T19:39:46.740Z]</span> <!--[31m--><span style="color: #CD0000;">Beginning multi-line color
</span></span><span class="pipeline-new-node" nodeId="9" enclosingId="6"><span style="color: #CD0000;">[2019-07-05T19:39:46.742Z] [Pipeline] echo
</span></span><span class="pipeline-node-9"><span class="timestamp"><b>12:39:46</b> </span><span style="color: #CD0000;"><span style="display: none">[2019-07-05T19:39:46.742Z]</span> More color
</span></span><span class="pipeline-new-node" nodeId="10" enclosingId="6"><span style="color: #CD0000;">[2019-07-05T19:39:46.744Z] [Pipeline] echo
</span></span><span class="pipeline-node-10"><span class="timestamp"><b>12:39:46</b> </span><span style="color: #CD0000;"><span style="display: none">[2019-07-05T19:39:46.745Z]</span> Ending multi-line color<!--[39m--></span>
</span>
```

Given that it's possible for `<span class="pipeline-new-node" [...]>` to immediately be followed by `<span style="color: [...]">`, I think it's necessary to keep this logic as-is.